x-pack/plugins/apm/server/routes/suggestions/get_suggestions_with_term.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { rangeQuery } from '../../../../observability/server';
+import { Setup } from '../../lib/helpers/setup_request';
+
+export async function getSuggestionsWithTerm({
+  field,
+  filter,
+  setup,
+  size,
+  string,
+  start,
+  end,
+}: {
+  field: string;
+  filter: QueryDslQueryContainer[];
+  setup: Setup;
+  size: number;
+  string: string;
+  start?: number;
+  end?: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search('get_suggestions_with_terms', {
+    apm: {
+      events: [
+        ProcessorEvent.transaction,
+        ProcessorEvent.error,
+        ProcessorEvent.metric,
```

Comment:
Does it make sense for a generic endpoint to have these indices hardcoded?

Reply:
I'm quite sure, I used the same as `/suggestions` endpoint.

Reply:
Okay, let's leave it for now. We might have to change this as we see more consumers.